npms/skale-observer/observer.mjs

```diff
@@ -0,0 +1,1078 @@
+    const cntSChains =
+        await opts.imaState.joSChainsInternal.callStatic.numberOfSchains(
+            { from: addressFrom } );
+    return cntSChains;
+}
+
+export function removeSChainDescDataNumKeys( joSChain ) {
+    const cnt = Object.keys( joSChain ).length;
+    for( let i = 0; i < cnt; ++ i ) {
+        try {
+            delete joSChain[i];
+        } catch ( err ) {
+        }
+    }
+}
+
+export async function loadSChain( addressFrom, idxSChain, hash, cntSChains, opts ) {
+    owaspUtils.ensureObserverOptionsInitialized( opts );
+    if( ! opts.imaState )
+        throw new Error( "Cannot load S-Chain description in observer, no imaState is provided" );
+    if( opts && opts.details ) {
+        if( log.verboseGet() >= log.verboseReversed().trace ) {
+            opts.details.write( cc.debug( "Loading S-Chain " ) + cc.notice( "#" ) +
+                cc.info( idxSChain + 1 ) + cc.debug( " of " ) + cc.info( cntSChains ) +
+                cc.debug( "..." ) + "\n" );
+        }
+    }
+    hash = hash ||
+        await opts.imaState.joSChainsInternal.callStatic.schainsAtSystem(
+            idxSChain, { from: addressFrom } );
+    if( opts && opts.details ) {
+        if( log.verboseGet() >= log.verboseReversed().trace )
+            opts.details.write( cc.debug( "    Hash " ) + cc.attention( hash ) + "\n" );
+    }
+    if( opts && opts.bStopNeeded )
+        return null;
+    let joData =
+        await opts.imaState.joSChainsInternal.callStatic.schains(
+            hash, { from: addressFrom } );
+    joData = owaspUtils.cloneObjectByRootKeys( joData );
+    const joSChain = { "data": joData };
+    removeSChainDescDataNumKeys( joSChain.data, addressFrom );
+    if( opts && opts.bStopNeeded )
+        return null;
+    await loadSChainParts( joSChain, addressFrom, opts );
+    if( opts && opts.details ) {
+        if( log.verboseGet() >= log.verboseReversed().trace ) {
+            opts.details.write( cc.debug( "    Desc " ) + cc.j( joSChain.data ) + "\n" );
+            opts.details.write( cc.success( "Done" ) + "\n" );
+        }
+    }
+    joSChain.isConnected = false;
+    return joSChain;
+}
+
+export async function loadSChains( addressFrom, opts ) {
+    owaspUtils.ensureObserverOptionsInitialized( opts );
+    if( ! opts.imaState )
+        throw new Error( "Cannot load S-Chains in observer, no imaState is provided" );
+    const cntSChains = await getSChainsCount( addressFrom, opts );
+    if( opts && opts.details ) {
+        if( log.verboseGet() >= log.verboseReversed().trace ) {
+            opts.details.write( cc.debug( "Have " ) + cc.info( cntSChains ) +
+                cc.debug( " S-Chain(s) to load..." ) + "\n" );
+        }
+    }
+    const arrSChains = [];
+    for( let idxSChain = 0; idxSChain < cntSChains; ++ idxSChain ) {
+        if( opts && opts.bStopNeeded )
+            break;
+        const joSChain = await loadSChain( addressFrom, idxSChain, null, cntSChains, opts );
```

Comment:
You can optimize it even more, right now for 20 chains it would be 81(1*getSchainsCount + 20*schainsAtSystem + 20*schains  20*getNodesInGroup + 20*multicall) calls.
I will recommend to add a function `getSchainsNodes(arrayOfSchains)` which would return array of array of nodes in these chains by multi call.
And in loadSChain you can pass an array of nodes for given schain.

So before this loop 1*`getSchainsCount` + 1*`multicall(schainsAtSystem+schains)` + 1*`getSchainsNodes(...)`
And then in the loop it would be 20 multicalls of nodes data.

So by this we would do 23 calls instead of 81 calls